BASE/src/Microsoft.ApplicationInsights/Channel/ITelemetryChannel.cs

```diff
@@ -31,5 +31,10 @@ public interface ITelemetryChannel : IDisposable
         /// Flushes the in-memory buffer.
         /// </summary>
         void Flush();
+
+        /// <summary>
+        /// Flushes the in-memory buffer asynchronously.
+        /// </summary>
+        Task<bool> FlushAsync();
```

Comment:
Two things:
1. Please document the return value.
2. Please add a `CancellationToken cancellationToken` parameter. That way callers can decide if/when to cancel (e.g. Ctrl+C, or critical shutdown, or based on a timeout)

Reply:
Also, is the intent to allow `FlushAsync` at any time, or only when the channel is shutting down? For example, what happens if you try to send more telemetry through the channel after `FlushAsync` completes successfully? Should you be able to call `FlushAsync` again? Or are those decisions up to each channel implementation?

Reply:
Added CancellationToken

Main intention of this FlushAsync is to get called when the channel is shutting down. But, this can be used in other scenarios to confirm a successful transmission.